### DISCUSSION TITLE
Github actions: kill all other jobs when one job fail

### DISCUSSION THREAD

**Discussion by w3ntao** [community/community#26544] [Actions]
<pre><code class="lang-auto">on: [push]

jobs:
  test:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        test_case: [a, b, c]
    steps:
      - uses: actions/checkout@v1
      - uses: actions/setup-python@v2
        with:
          python-version: '3.x'
      - run: python3 script.py ${{ matrix.test_case }}
</code></pre>
<p>This is my <code>.github/workflow/main.yml</code> to trigger test_case <code>a</code> , <code>b</code> , <code>c</code> concurrently. It is considered success only when all 3 jobs passed, and otherwise failure.</p>
<p>So, to save my time, how do I send a signal to kill other jobs when one of them fails?</p>

**Comment by otter-computer** [comment, marked as answer]
<p>This should already be the case as <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast"><code>fail-fast</code> is set to <code>true</code> by default</a> when defining a matrix.</p>

**Comment by indiVar0508** [comment]
Hi 
I am also facing similar issue, 
i have a jobs that look like

```
jobs
  job_a:
         ....

  job_b:
        ....
  
  job_c:
        ....

  job_d:
        ....
```
these jobs are in same workflow file, if a single job fails i want to cancel all of them.
how can i do same?

**Comment by duaneking** [comment]
Do we have a fix?

If the expectation of people working at Github is that a failed task in part of a job of a pipeline means that all other jobs in the pipeline fails due to that failed job, I'm here to say "no that's not what we see in real life".

It is very easy to create a multi job continuous integration run using multiple platforms (Linux and windows is fine) and have one of them fail while the other ones just keep chugging on even though they should have just stopped by now because their peers failed.
